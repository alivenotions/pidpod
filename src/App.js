import React, { Component } from 'react'

class App extends Component {
  render() {
    return <div className="App">Pidpod</div>
  }
}

export default App
