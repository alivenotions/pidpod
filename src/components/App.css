@import url('https://fonts.googleapis.com/css?family=Fira+Sans:400,500,700');

body {
  margin: 0;
  padding: 0;
  font-family: 'Fira Sans';
  background: #eff2ef;
}
