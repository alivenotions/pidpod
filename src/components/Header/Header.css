@import url('https://fonts.googleapis.com/css?family=Pacifico');

header {
  padding: 1em;
  margin: 0 1.5em;
  border-bottom: 1px solid #dde0dd;
}

header h1 {
  display: inline;
  font-family: 'Pacifico', sans-serif;
  padding-left: 15%;
  font-size: 3em;
}

header .add-podcast {
  margin: auto 0;
  position: absolute;
  right: 1em;
  top: 3em;
}

header button {
  font-size: 1em;
  padding: 0.4em;
  margin: 0 0.5em;
  font-weight: bold;
  height: 2.5em;
  width: 2.5em;
  border-radius: 50%;
  background: linear-gradient(45deg, #dde0dd 25%, #c4c4c4);
  box-shadow: 3px 1.5px 1px #cecfce;
  border: 1px solid #adafad;
  cursor: pointer;
}

header button:active {
  background: linear-gradient(45deg, #c4c4c4 25%, #dde0dd);
}

header button:disabled {
  background: #bebebe;
  opacity: 0.7;
  cursor: default;
  color: #807f7f;
  box-shadow: 0 0 0;
}

header input {
  font-size: 1em;
  /* width: 0; */
  padding: 0.4em;
  margin: 0 0.5em;
}
