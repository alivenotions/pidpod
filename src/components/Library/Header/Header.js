import React from 'react'

class Header extends React.Component {
  addPodcast = event => {
    event.preventDefault()
    // FIXME: Create a modal instead of an input type
    const rssUrl = document.getElementById('rssUrl').value
    console.log(rssUrl)
    this.props.addPodcast(rssUrl)
  }

  render() {
    return (
      <header>
        <h1>{this.props.heading}</h1>
        {/* TODO: Abstract this into a button component */}
        <input type="text" id="rssUrl" />
        <button onClick={this.addPodcast}>Add Podcast</button>
      </header>
    )
  }
}

export default Header
