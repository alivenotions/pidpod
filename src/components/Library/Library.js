import React, { Fragment } from 'react'
import { notify, notifyError } from '../../utils/toastNotifs'
import { getRSSParser } from '../../utils/rssParser'

class Library extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      podcastFeeds: [],
    }
  }

  componentDidUpdate = prevProps => {
    if (this.props.rssUrl !== prevProps.rssUrl) {
      this.addPodcastRSSFeedFromUrl(this.props.rssUrl)
    }
  }

  addPodcastRSSFeedFromUrl = rssUrl => {
    getRSSParser()
      .parseURL(rssUrl)
      .then(feed => {
        this.setState({ podcastFeeds: [...this.state.podcastFeeds, feed] })
        notify('Successfully added the podcast to Library')
      })
      .catch(_ => notifyError('The URL seems to be wrong!'))
  }

  render() {
    return <Fragment>{this.props.heading}</Fragment>
  }
}

export default Library
