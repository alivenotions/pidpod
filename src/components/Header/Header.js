import React, { Component } from 'react'

import './Header.css'

class Header extends Component {
  addPodcast = event => {
    event.preventDefault()
  }

  handleAddPodcastClick = event => {}

  render() {
    return (
      <header>
        <h1>{this.props.title}</h1>
        <div className="add-podcast">
          <input placeholder="Add RSS Feed" />
          <button>+</button>
        </div>
      </header>
    )
  }
}

export default Header
