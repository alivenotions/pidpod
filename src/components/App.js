import React, { Component, Fragment } from 'react'

import Header from './Header/Header'
import Library from './Library/Library'
import './App.css'

class App extends Component {
  state = {
    lastRSSUrl: '',
  }

  setLastRSSUrl = url => {
    this.setState({ lastRSSUrl: url })
  }

  render() {
    return (
      <Fragment>
        <Header title="Pidpod" addPodcast={this.setLastRSSUrl} />
        <Library heading="My Podcasts" rssUrl={this.state.lastRSSUrl} />
      </Fragment>
    )
  }
}

export default App
