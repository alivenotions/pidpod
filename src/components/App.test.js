import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import App from './App'
import Header from './Header/Header'
import Library from './Library/Library'

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
  })

  it('renders a <Header /> component', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find(Header)).toHaveLength(1)
  })

  it('renders a <Library /> component', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find(Library)).toHaveLength(1)
  })
})
