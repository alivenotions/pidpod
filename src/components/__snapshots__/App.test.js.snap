// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> passes the correct rssUrl according to state 1`] = `
<Fragment>
  <Header
    addPodcast={[Function]}
    title="Pidpod"
  />
  <Library
    heading="My Podcasts"
    rssUrl="mock.com/url/rss"
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</Fragment>
`;

exports[`<App /> renders correctly 1`] = `
<Fragment>
  <Header
    addPodcast={[Function]}
    title="Pidpod"
  />
  <Library
    heading="My Podcasts"
    rssUrl=""
  />
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
</Fragment>
`;
